compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -321,6 +323,32 @@ impl<'a, 'b, 'tcx> TypeFolder<'tcx> for AssocTypeNormalizer<'a, 'b, 'tcx> {
         self.selcx.tcx()
     }
 
+    #[instrument(level = "debug", skip(self))]
+    fn fold_binder<T>(&mut self, t: ty::Binder<'tcx, T>) -> ty::Binder<'tcx, T>
```

Comment:
I think the comment in `fold_ty` below should be updated (since we now instantiate bound regions here, instead of 'eventually').